src/config.py

```diff
@@ -0,0 +1,63 @@
+import os
+from typing import TypeVar
+
+from dotenv import load_dotenv
+
+# Generic type, used for static type checking
+T = TypeVar("T")
+
+load_dotenv()
+
+
+def get_env_variable(var_name: str, default: T = None, allow_empty: bool = False, cast_to: T = str) -> T:
+    """
+     Reads an environment variable, with settings to allow defaults, empty values, and type casting
+
+    Parameters
+    ----------
+    var_name : str
+        The name of the environment variable to retrieve.
+    default : T = None
+        Default return value if the environment variable does not exist. Doesn't override empty string vars.
+    allow_empty : bool
+        If False then a KeyError will be raised if the environment variable is empty.
+    cast_to : T
+        The type that the variable is cast to when returned e.g. int or bool.
+
+    :return: The environment variable, or default if it does not exist.
+    :raises KeyError if allow_empty is False and the environment variable is empty string or None
+    :raises ValueError if cast_to is not compatible with the value stored.
+    """
```

Comment:
As discussed, it would be a good idea to use the same docstring format, i.e. follow the numpy docstring style guide https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard.

Reply:
Fixed